piet-direct2d/src/text.rs

```diff
@@ -305,25 +305,23 @@ impl TextLayout for D2DTextLayout {
     }
 
     // Can panic if text position is not at a code point boundary, or if it's out of bounds.
-    fn hit_test_text_position(&self, text_position: usize) -> Option<HitTestPosition> {
+    fn hit_test_text_position(&self, idx: usize) -> Option<HitTestPosition> {
+        let idx = idx.min(self.text.len());
+        assert!(self.text.is_char_boundary(idx));
         // Note: Directwrite will just return the line width if text position is
         // out of bounds. This is what want for piet; return line width for the last text position
         // (equal to line.len()). This is basically returning line width for the last cursor
         // position.
 
         // Now convert the utf8 index to utf16.
         // This can panic;
-        let idx_16 = util::count_utf16(&self.text[0..text_position]);
-        let line = util::line_number_for_position(&self.line_metrics, text_position);
-        // panic or Result are also fine options for dealing with overflow. Using Option here
-        // because it's already present and convenient.
-        // TODO this should probably go before convertin to utf16, since that's relatively slow
-        let idx_16 = idx_16.try_into().ok()?;
-
-        let trailing = false;
```

Comment:
Very minor style nit, but I feel it's useful for this to continue to be a var; a bool by itself (as opposed to a two-state enum value) feels like it should have some documentation in the code.